### PR TITLE
Ignore the home field when adding a config map to viper.

### PR DESCRIPTION
**Pull request by SpicyLemon**
## Description

In the config manager, when loading a config map into viper, do not include the "home" field.

The Tendermint BaseConfig struct has this field:
```
RootDir string `mapstructure:"home"`
```

So when we convert a `tendermint.Config` into a field value map, it will have a "home" entry. The `DefaultConfig()` has it as `""`, and it's not a field in the tendermint config file. So it will almost always be `""`. Then, when we provide the field value map to viper, viper thinks, "I have been given `""` as the `"home"` value, so I don't need to use the default (defined with the command's flags)." Later, whenever anyone asks viper to `.Get(flags.FlagHome)`, an empty string is returned. 

I verified this by:
1. Adding some `fmt.Printf("home dir: \"%v\"\n", vpr.Get(flags.FlagHome))` lines before and after tm config stuff in `loadUnpackedConfig`.
2. Deleting the `data/` dir from my local default config location (`/Users/danielwedul/Library/Application Support/Provenance`).
3. Running `provenanced start` in an environment that doesn't have a `PIO_HOME` defined.

Before the fix in this PR:
1. Those log lines showed a `""` after calling `addFieldMapToViper` with the default tendermint config struct.
2. The app panicked with `panic: mkdir data/snapshots: no such file or directory` (where `.Get(flags.FlagHome)` is called).

After the fix in this PR:
1. All those log lines show the correct home dir.
2. The app stops with a message of `open /Users/danielwedul/Library/Application Support/Provenance/data/priv_validator_state.json: no such file or directory` which is expected since I had deleted that `data` dir.
3. The app creates that `data/snapshots` directory as expected.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -75,11 +76,11 @@ The `paua` upgrade will increase all validators' max commission to 100% and max
   [PR 1334](https://github.com/provenance-io/provenance/pull/1334),
   [PR 1348](https://github.com/provenance-io/provenance/pull/1348),
   [PR 1317](https://github.com/provenance-io/provenance/pull/1317),
-  [PR 1278](https://github.com/provenance-io/provenance/pull/1278).  
+  [PR 1278](https://github.com/provenance-io/provenance/pull/1278). \
   See the following `RELEASE_NOTES.md` for details:
   [v0.46.10-pio-1](https://github.com/provenance-io/cosmos-sdk/blob/v0.46.10-pio-1/RELEASE_NOTES.md),
   [v0.46.8-pio-3](https://github.com/provenance-io/cosmos-sdk/blob/v0.46.8-pio-3/RELEASE_NOTES.md),
-  [v0.46.7-pio-1](https://github.com/provenance-io/cosmos-sdk/blob/v0.46.7-pio-1/RELEASE_NOTES.md).  
+  [v0.46.7-pio-1](https://github.com/provenance-io/cosmos-sdk/blob/v0.46.7-pio-1/RELEASE_NOTES.md). \
```
Comment (SpicyLemon) [inline CHANGELOG.md:83]: In markdown, a trailing <space><space> indicates a new line. However, our `make linkify` script trims trailing spaces, causing these to be lost.
Luckily, in most markdown, you can use a `\` to indicate a new line too.

So that's why this change is here.

**Comment by SpicyLemon** [comment]
In markdown, a trailing <space><space> indicates a new line. However, our `make linkify` script trims trailing spaces, causing these to be lost. Luckily, in [most] markdown, you can use a `\` to indicate a new line too. So that's why the extra couple lines are changed in the `CHANGELOG.md`.

**Review by iramiller — APPROVED** [review]
Nice Find!

**Review by egaxhaj — APPROVED** [review]
